### PR TITLE
exclude test directories from analysis

### PR DESCRIPTION
**Pull request by devoncarew**
- exclude test directories from analysis
- reset us down to zero analysis warnings (including deprecation warnings)

### CODE REVIEW

**Comment by devoncarew** [comment]
@sethladd

**Comment by sethladd** [comment]
Can you make this PR just the `.analysis_options` ?

My other PR removes the dart:profiler related code.

File: test_package/.analysis_options [hunk 1]
```diff
@@ -0,0 +1 @@
+{ analyzer: { exclude: ['**'] } }
```
Comment (sethladd) [inline test_package/.analysis_options]: huh, I thought this file was a YAML file ?

Comment (devoncarew) [inline test_package/.analysis_options]: yaml's a superset of json :)

Comment (sethladd) [inline test_package/.analysis_options]: +1 for the YAML format. It's easier to read.

If there's a reason to use the JSON version, that's fine, too.

Comment (devoncarew) [inline test_package/.analysis_options]: I think of this file as just a 1-line marker file - don't look in this directory.

Hopefully the AS issue I filed will be addressed - the parent pubspec can exclude this directory, and this file can just be deleted.

**Pull request merged**